superset/utils/core.py

```diff
@@ -1401,19 +1402,30 @@ def get_column_names_from_metrics(metrics: List[Metric]) -> List[str]:
     return columns
 
 
-def serialize_pandas_dtypes(dtypes: List[np.dtype]) -> List[GenericDataType]:
-    """Serialize pandas/numpy dtypes to JavaScript types"""
-    mapping = {
-        "object": GenericDataType.STRING,
-        "category": GenericDataType.STRING,
-        "datetime64[ns]": GenericDataType.TEMPORAL,
-        "int64": GenericDataType.NUMERIC,
-        "in32": GenericDataType.NUMERIC,
-        "float64": GenericDataType.NUMERIC,
-        "float32": GenericDataType.NUMERIC,
-        "bool": GenericDataType.BOOLEAN,
+def extract_dataframe_dtypes(df: pd.DataFrame) -> List[GenericDataType]:
+    """Serialize pandas/numpy dtypes to generic types"""
+
+    # omitting string types as those will be the default type
+    inferred_type_map: Dict[str, GenericDataType] = {
+        "floating": GenericDataType.NUMERIC,
+        "integer": GenericDataType.NUMERIC,
+        "mixed - integer": GenericDataType.NUMERIC,
+        "mixed - integer - float": GenericDataType.NUMERIC,
```

Comment:
Should be `mixed-integer` and `mixed-integer-float`. And `mixed-integer` should probably be string since `infer_dtype(['a','b','c',1])` and `infer_dtype(['a','b','c',1, 1.1])` are also `mixed-integer`.

Reply:
Right you are - I copy-pasted them incorrectly and forgot to add tests for these cases, will do that now.